airflow/providers/sftp/operators/sftp_batch.py

```diff
@@ -0,0 +1,294 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""This module contains SFTP Batch operator."""
+import os
+import re
+from pathlib import Path
+from typing import Any, List
+
+from paramiko.sftp_client import SFTPClient
+
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+from airflow.providers.sftp.operators.sftp import SFTPOperation, _check_conn, _make_intermediate_dirs
```

Comment:
WDYT about moving `SFTPOperation `, `_check_conn` and `_make_intermediate_dirs` into a centralized location in the provider since these are used across multiple operators now?

Perhaps `_check_conn` and `_make_intermediate_dirs` can live in a `providers/sftp/utils/` directory? A pattern in multiple providers for classes like `SFTPOperation` is to store them in the corresponding hook.

Reply:
Actually, maybe the move is to have both the `SFTPOperator` and `SFTPBatchOperator` both live in a single `providers/sftp/operators/sftp.py` file.

Reply:
I think I'll make a separate file for common functions.

`Actually, maybe the move is to have both the SFTPOperator and SFTPBatchOperator both live in a single providers/sftp/operators/sftp.py file.`
I do not like large files with code where classes do not depend on each other

Reply:
done